theano/scalar/basic.py

```diff
@@ -3659,7 +3659,7 @@ def __init__(self, inputs, outputs):
         self.nin = len(inputs)
         self.nout = len(outputs)
         self.init_fgraph()       # self.fgraph
-
+        self.init_c_code()
```

Comment:
I know this is a hack, but I think we shouldn't have the `_c_code` as a key to check the equality. Do you agree @nouiz ?